src/js/content/content.spec.js

```diff
@@ -14,29 +14,95 @@
-    }));
-  });
-
-  it('getData should send message, and shouldn not fetch any datas for html data source', async () => {
-    await getData('text/html');
-    expect(chrome.runtime.sendMessage.calledOnce).toEqual(true);
-    expect(fetch).toHaveBeenCalledTimes(0);
-    chrome.flush();
-  });
-
-  it('getData should send message, and fetch data for json data source', async () => {
-    await getData('application/json');
-    expect(chrome.runtime.sendMessage.calledOnce).toEqual(true);
-    expect(fetch).toHaveBeenCalledTimes(1);
-    expect(fetch).toHaveBeenCalledWith(window.location.href);
-    chrome.flush();
-  });
+function setupFetchMock(result) {
+  global.fetch = jest.fn(() => Promise.resolve({
+    json: () => Promise.resolve(result),
+  }));
+}
+
+function setupHtmlMock(result) {
+  nodesHelper.findFragmentsInContent = jest.fn(() => result);
+}
+
+beforeEach(() => {
+  global.chrome = { runtime: { sendMessage: jest.fn() } };
+});
+
+it('getData passes no fragments for unsupported response type', async () => {
+  await getData('text/plain');
+
+  expect(chrome.runtime.sendMessage).toHaveBeenCalledWith({ fragmentsData: [] }, expect.any(Function));
+});
+
+it('getData passes one fragment with no nodes for JSON object in response', async () => {
+  const mockedResponse = { _knotx_fragment: { id: '0' }, some_data: 42 };
+  setupFetchMock(mockedResponse);
+
+  await getData('application/json');
+
+  const expectedMessage = {
+    fragmentsData: [
+      {
+        // eslint-disable-next-line no-underscore-dangle
+        debug: mockedResponse._knotx_fragment,
+        nodes: [],
+      },
+    ],
+  };
+
+  expect(chrome.runtime.sendMessage).toHaveBeenCalledWith(expectedMessage, expect.any(Function));
+});
+
+it('getData passes no fragments for empty JSON array in response', async () => {
+  const mockedResponse = [];
+  setupFetchMock(mockedResponse);
+
+  await getData('application/json');
+
+  const expectedMessage = { fragmentsData: [] };
+
+  expect(chrome.runtime.sendMessage).toHaveBeenCalledWith(expectedMessage, expect.any(Function));
+});
+
+it('getData passes multiple fragments with no nodes for JSON array in response', async () => {
+  const mockedResponse = [
+    { _knotx_fragment: { id: '0' }, some_data: 42 },
+    { _knotx_fragment: { id: '1' }, some_other_data: 'lorem' },
+  ];
+  setupFetchMock(mockedResponse);
+
+  await getData('application/json');
+
+  const expectedMessage = {
+    fragmentsData: [
+      {
+        // eslint-disable-next-line no-underscore-dangle
+        debug: mockedResponse[0]._knotx_fragment,
+        nodes: [],
+      },
+      {
+        // eslint-disable-next-line no-underscore-dangle
+        debug: mockedResponse[1]._knotx_fragment,
+        nodes: [],
+      },
+    ],
+  };
+
+  expect(chrome.runtime.sendMessage).toHaveBeenCalledWith(expectedMessage, expect.any(Function));
+});
+
+it('getData delegates processing to nodeHelper for HTML response', async () => {
```

Comment:
Should we handle the same cases for HTML and JSON? How about parametrized tests?

Reply:
They are quite different, given that the JSON fragments don't use nodes while HTML ones do. Test suite for HTML fragments already exists and is more complex then the JSON one needs to be. I think it would just add unnecessary complexity